src/main/scala/scalismo/mesh/TetrahedralList.scala

```diff
@@ -0,0 +1,116 @@
+
+/*
+ * Copyright University of Basel, Graphics and Vision Research Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package scalismo.mesh
+
+import scalismo.common.PointId
+
+/** List of tetrahedrons used to define the tetrahedralization in the [[TetrahedralMesh]]. */
+case class TetrahedralList(tetrahedrons: IndexedSeq[TetrahedralCell]) {
+
+  val pointIds = extractPointIds(tetrahedrons)
+  val tetrahedronIds = tetrahedrons.indices.map(i => TetrahedronId(i))
+
+  def tetrahedron(id: TetrahedronId): TetrahedralCell = tetrahedrons(id.id)
+
+  /** Function returning a list of all ids of tetrahedrons containing the point id. */
+  lazy val adjacentTetrahedronsForPoint: PointId => IndexedSeq[TetrahedronId] = {
+    // list structure
+    val emptyMapData = for (p <- pointIds) yield p -> collection.mutable.Set.empty[TetrahedronId]
```

Comment:
I think it is not as easy as your suggestion. You would need a MultiSet as a Set is not enough. Otherwise, you would have a list of tuples, where the first entry occurs multiple times. Then you would need to form a list with the second entries for all distinct first entries. If you have something efficient and readable alternative, I am glad to see your suggestion.